Automattic/jetpack
# Social: Added Bluesky to social previews

## Proposed changes:

* Social: Added Bluesky to social previews

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Ensure that your blog has the `jetpack-social-bluesky-connection` blog sticker
* Run `pnpm install` and `jetpack build plugins/jetpack plugins/social`
* Goto post editor
* Add a Bluesky connection
* Click on "Preview social posts"
* Confirm that you see the Bluesky preview
* Confirm that it matches the previews on Bluesky with and without custom media
* Click on "Preview" in Social Previews panel
<img width="285" alt="image" src="https://github.com/user-attachments/assets/be86d846-23d6-4264-97bf-cf7bbd18dbfd">

* Goto Bluesky preview
* Confirm that you see the Bluesky preview for Your post and Link preview sections
* Confirm that both look fine
* Now close the modal and remove the Bluesky connection
* Click on "Preview" again
* Confirm that you don't see the "Your post" section

<table>
    <tbody>
        <tr>
            <th>Has featured image</th>
            <th>Has custom media</th>
            <th>Result</th>
        </tr>
        <tr>
            <th colspan="3">Social post preview</th>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ✅ </td>
            <td> <img width="932" alt="Screenshot 2024-10-07 at 1 44 08 PM"
                    src="https://github.com/user-attachments/assets/3502ce0a-0d29-48fd-ab5e-197f0b659c73">
            </td>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ❌ </td>
            <td> <img width="935" alt="Screenshot 2024-10-07 at 1 44 17 PM"
                    src="https://github.com/user-attachments/assets/6f465fda-87e3-4d7f-9c0f-e3a280ed98a6">
            </td>
        </tr>
        <tr>
            <td> ❌ </td>
            <td> ❌ </td>
            <td> <img width="935" alt="Screenshot 2024-10-07 at 2 09 47 PM"
                    src="https://github.com/user-attachments/assets/75f87c4c-0c34-4895-b216-c71838903a3e">
            </td>
        </tr>
        <tr>
            <th colspan="3">Common Social preview</th>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ✅ </td>
            <td> <img width="959" alt="Screenshot 2024-10-07 at 1 45 58 PM"
                    src="https://github.com/user-attachments/assets/6e582e63-6aad-43dc-b9c2-81ab6b64bdd8">
            </td>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ✅ </td>
            <td> <img width="963" alt="Screenshot 2024-10-07 at 1 46 12 PM"
                    src="https://github.com/user-attachments/assets/e08f296a-7e8e-4b05-9ef4-532bb989b8cf">
            </td>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ❌ </td>
            <td> <img width="957" alt="Screenshot 2024-10-07 at 1 44 54 PM"
                    src="https://github.com/user-attachments/assets/26445605-97d4-4c68-835a-9edf1e703098">
            </td>
        </tr>
        <tr>
            <td> ✅ </td>
            <td> ❌ </td>
            <td> <img width="965" alt="Screenshot 2024-10-07 at 1 45 15 PM"
                    src="https://github.com/user-attachments/assets/d540c48c-1243-4b47-9c62-c97a94bcc98e">
            </td>
        </tr>
        <tr>
            <td> ❌ </td>
            <td> ❌ </td>
            <td> <img width="951" alt="Screenshot 2024-10-07 at 2 10 05 PM"
                    src="https://github.com/user-attachments/assets/6076998b-e653-4449-a96d-8856622df10c">
            </td>
        </tr>
    </tbody>
</table>
